From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Vandomas <vandcim@gmail.com>
Date: Thu, 27 Aug 2026 15:58:43 +0300
Subject: [PATCH 21/21] zink: do not use implicit sync on kosmickrisp

kosmickrisp is a mesa driver, it just never got added to the list. Metal
has no implicit sync to work around anyway, so the empty submit and the
fence wait before every present buy nothing here.

Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/44048>
(cherry picked from commit e9a87f80b2d6df916fffcd7798fdf9b7c57f3d58)
---
 src/gallium/drivers/zink/zink_screen.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/gallium/drivers/zink/zink_screen.c b/src/gallium/drivers/zink/zink_screen.c
index c11c6a2cabe..1d76286ce3b 100644
--- a/src/gallium/drivers/zink/zink_screen.c
+++ b/src/gallium/drivers/zink/zink_screen.c
@@ -2921,6 +2921,7 @@ init_driver_workarounds(struct zink_screen *screen)
    case VK_DRIVER_ID_MESA_V3DV:
    case VK_DRIVER_ID_MESA_PANVK:
    case VK_DRIVER_ID_MESA_NVK:
+   case VK_DRIVER_ID_MESA_KOSMICKRISP:
    case VK_DRIVER_ID_QUALCOMM_PROPRIETARY:
       screen->driver_workarounds.implicit_sync = false;
       break;
